source/base/quadrature_lib.cc

```diff
@@ -943,6 +943,8 @@ QSorted<dim>::QSorted(const Quadrature<dim> &quad) :
     {
       this->weights[i]           = quad.weight(permutation[i]);
       this->quadrature_points[i] = quad.point(permutation[i]);
+      if (permutation[i] != i)
+        this->is_tensor_product_flag = false;
```

Comment:
I don't want to say that a `Quadrature` object becomes a tensor product if the permutation is the identity. Instead, a tensor product that is permuted should not be considered a tensor product anymore. I clarified this in the comment above this loop.